Modules/xASL_module_ASL.m

```diff
@@ -125,358 +105,13 @@
-if nTE~=nVolumes
-    warning(['Number of echo times (n=' xASL_num2str(nTE) ') should be equal to number of ASL volumes (n=' xASL_num2str(nVolumes) ')']);
-end
-
-
-%% D4. Multi-PLD parsing
-if ~isfield(x.Q,'Initial_PLD')
-    warning('Missing field x.Q.Initial_PLD, this is needed for ASL quantification');
-	x.Q.bQuantifyMultiPLD = false;
-elseif isempty(x.Q.Initial_PLD) || ~isnumeric(x.Q.Initial_PLD)
-    warning('Illegal field x.Q.Initial_PLD, this should not be empty and should contain numerical values');
-	x.Q.bQuantifyMultiPLD = false;
-else
-    x.Q.uniqueInitial_PLD = unique(x.Q.Initial_PLD);
-    x.Q.NumberPLDs = length(x.Q.uniqueInitial_PLD);
-    
-    if sum(x.Q.Initial_PLD<=0)>0
-        warning('x.Q.Initial_PLD should be positive');
-    end
-
-    % Handle different number of PLDs
-    if x.Q.NumberPLDs==1
-        fprintf('%s\n', 'Single-PLD ASL detected');
-		x.Q.bQuantifyMultiPLD = false;
-    elseif x.Q.NumberPLDs>1
-		if isfield(x.Q, 'bQuantifyMultiPLD') && ~x.Q.bQuantifyMultiPLD
-			fprintf('%s\n', 'Multi PLDs detected, but multi-PLD quantification set OFF.');
-		else
-			% In case it wasn't defined or it was defined as true, we can set it to true
-			x.Q.bQuantifyMultiPLD = true;
-		end
-		if x.Q.bQuantifyMultiPLD
-			fprintf('%s\n', 'Multi PLDs detected, processing this as multi-PLD ASL');
-			fprintf('%s\n', 'Note that this feature is still under development');
-		end
-    end
-
-    fprintf('%s\n', 'Detected the following unique PLDs (ms):')
-    for iPLD = 1:x.Q.NumberPLDs-1
-        fprintf('%d, ',x.Q.uniqueInitial_PLD(iPLD));
-    end
-    fprintf('%d\n',x.Q.uniqueInitial_PLD(end));
-end
-
-% Deal with too short vectors (e.g., repetitions in the case of single-PLD)
-nPLD = length(x.Q.Initial_PLD);
-if mod(nVolumes, nPLD)~=0
-    error(['Number of PLDs (n=' xASL_num2str(nPLD) ') does not fit in nVolumes (n=' xASL_num2str(nVolumes) ')']);
-    % If we don't issue an error here, the repmat will crash
-end
-
-factorPLD = nVolumes/nPLD;
-x.Q.Initial_PLD = repmat(x.Q.Initial_PLD(:), [factorPLD 1]);
-nPLD = length(x.Q.Initial_PLD);
-
-% Number of initial PLDs should be equal to number of ASL volumes
-if nPLD~=nVolumes
-    warning(['Number of PLDs (n=' xASL_num2str(nPLD) ') should be equal to number of ASL volumes (n=' xASL_num2str(nVolumes) ')']);
-end
-
-%% D5. Labeling Duration parsing
-
-if ~isfield(x.Q,'LabelingDuration')
-    warning('Missing field x.Q.LabelingDuration, this is needed for ASL quantification');
-elseif isempty(x.Q.LabelingDuration) || ~isnumeric(x.Q.LabelingDuration)
-    warning('Illegal field x.Q.LabelingDuration, this should not be empty and should contain numerical values');
-else
-    x.Q.uniqueLabelingDuration = unique(x.Q.LabelingDuration);
-    x.Q.NumberLDs = length(x.Q.uniqueLabelingDuration);
-    
-    if sum(x.Q.LabelingDuration<=0)>0
-        warning('x.Q.LabelingDuration should be positive');
-    end
-
-    % Handle different number of Labeling Durations
-    if x.Q.NumberLDs==1
```

Comment:
bQUantifyMultiLD is not really used later - as this is a special case of MultiPLD.

So please take these warnings and rewrite them for the case of bQuantifyMultiLD and bQuantifyMultiPLD.

Reply:
Don't know what you mean by this.